test/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+cmake_minimum_required ( VERSION 3.13.0 )
```

Comment:
Why are we moving out to the test/ folder?

Reply:
This PR adds a mechanism for testing the `MQTT_DO_NOT_USE_CUSTOM_CONFIG` macro (that allows building the MQTT library _as-is_ without a custom config) through the library target.